substrate/frame/bounties/src/lib.rs

```diff
@@ -551,8 +551,8 @@ pub mod pallet {
 						T::Currency::reserve(curator, deposit)?;
 						bounty.curator_deposit = deposit;
 
-						let update_due = frame_system::Pallet::<T>::block_number() +
-							T::BountyUpdatePeriod::get();
+						let update_due =
+							Self::treasury_block_number() + T::BountyUpdatePeriod::get();
```

Comment:
I imagine most of the logic breaks when a parachain migrates from `System` to the `Relay` `BlockNumberProvider`...  
Not sure what can be done about it without re-writing the pallet to use timepoints. Maybe we can use this as first step and advise to still use `System`.

Reply:
I think in the long term we need some kind of block-number-abstraction like https://github.com/paritytech/polkadot-sdk/issues/3268, but its a bit stale now as low-prio. That is not necessarily timestamps, but could be implemented with Relay blocknumber or something else.